### PR TITLE
[`#32`] Add a Changelog file

### LINKED ISSUE #32: Add a `CHANGELOG` file
Add a changelog file. What is a changelog file:
* https://www.youtube.com/watch?v=Pi8akLplC9M&ab_channel=NickJanetakis
* https://keepachangelog.com/en/1.0.0/

### PR DESCRIPTION
**Pull request by Allstreamer**
- Added `CHANGELOG.md` based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) template
- Added Dev-branch changelogs

### CODE REVIEW

**Comment by John15321** [comment]
Technically we had our `0.0.1` release https://crates.io/crates/rust-pip so we have to that into account somehow

**Comment by Allstreamer** [comment]
I'll include the main branch commits as 0.0.1

**Comment by Allstreamer** [comment]
You really pay attention to the smallest details

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [0.0.1] 22 July 2022
+- Added Basic CLI functionality using StructOpt
+- Added Base Project
+
+## [Unreleased]
```
Comment (John15321) [inline CHANGELOG.md]: Please make sure that the Unreleased stays on top

**Comment by John15321** [comment]
Markdown has its own formatting guidelines and given that it's a markup language that is whitespace sensitive it's important to make sure that it's formatted. If you are using VSCode you can use the `markdownlint` extension for that. But now that I think of it we could use a CI job to check that our markdown docs are properly formatted. You can make a task about it and even add it in this PR if you want:
image

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Changelog
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+### Added
```
Comment (John15321) [inline CHANGELOG.md]: Also, a very good changelog you can use as a reference is this one:
https://github.com/python-poetry/poetry/blob/master/CHANGELOG.md?plain=1

E.G. something about documentation (like adding a changelog) should be under its own `Docs` section

Comment (Allstreamer) [reply CHANGELOG.md]: @John15321 Formatting should now be valid

Comment (Allstreamer) [reply CHANGELOG.md]: @John15321 Resolved issues raised

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+### Changed
+
+- Changed CLI to use CLAP
```
Comment (John15321) [inline CHANGELOG.md]: Everything looks good. Just make sure that PRs etc are linked 

Example: https://github.com/python-poetry/poetry/blob/1.2.0rc1/CHANGELOG.md?plain=1

Comment (Allstreamer) [reply CHANGELOG.md]: @John15321 Not all commits had pull requests, but the ones that did i added in.

Comment (cnpryer) [reply CHANGELOG.md]: Is poetry autogenerating their content using GitHub releases?

Comment (Allstreamer) [reply CHANGELOG.md]: If you are talking about their release changelog then yes, I believe so.

Comment (John15321) [reply CHANGELOG.md]: > Is poetry autogenerating their content using GitHub releases?

The normal changellg, they don't

Comment (John15321) [reply CHANGELOG.md]: > Is poetry autogenerating their content using GitHub releases?

In the normal changellg, they don't

**Review by John15321 — APPROVED** [review]
git gut

**Pull request merged**